jet_tracking/datastream.py

```diff
@@ -122,370 +163,414 @@ def read_value(self):  # needs to initialize first maybe using a decorator?
-        self.buffers = {"i0": deque([0], self.buffer_size),
-                        "diff": deque([0], self.buffer_size),
-                        "ratio": deque([0], self.buffer_size),
-                        "time": deque([0], self.buffer_size)}
+        This method starts the StatusThread by setting the paused attribute to False.
+        """
+        log.info("Inside of the start_it method of StatusThread.")
+        self.paused = False
 
-    def connect_signals(self):
-        self.signals.mode.connect(self.update_mode)
-        self.signals.changeDisplayFlag.connect(self.change_display_flag)
-        self.signals.changePercent.connect(self.set_percent)
-        self.signals.changeCalibrationSource.connect(
-            self.set_calibration_source)
-        self.signals.enableTracking.connect(self.tracking)
-        self.signals.valuesRequest.connect(self.send_info_to_motor)
+    def stop_it(self, abort):
+        """
+        Stop the StatusThread.
 
-    @staticmethod
-    def fix_size(old_size, new_size, b):
-        d = new_size - old_size
-        if d <= 0:
-            for key in b:
-                old = list(b[key])
-                new = old[-d:]
-                b[key] = deque(new, new_size)
+        This method stops the StatusThread by setting the paused attribute to True. If abort is True, the thread will be interrupted.
+
+        Parameters:
+        ----------
+        abort : bool
+            A flag indicating whether the thread should be interrupted.
+        """
+        self.paused = True
+        if abort:
+            self.thread().requestInterruption()
         else:
-            for key in b:
-                old = list(b[key])
-                b[key] = deque(old, new_size)
-        return b
+            pass
 
-    @staticmethod
-    def append_or_set(choice, d, vals, idx):
+    def run_data_thread(self):
         """
-        Appends or assigns values to a dictionary.
+        Run the data thread.
 
-        Send in a list of vals in the order 'i0', 'diff', 'ratio' and 'time'
-        and the choice to either "append" or "set" and the dictionary d and
-        then execute the action.
+        This method is the long-running task that collects data points. It reads values from the ValueReader,
+        updates buffers, checks status updates, calculates averages, and emits signals to refresh graphs.
 
-        Keyword arguments:
-        choice -- either "append" or "set"
-        d -- the dictionary that will be updated
-        vals -- the values or value to be added
-        idx -- the current x_cycle value
+        Note: This method runs in a continuous loop until the thread is paused.
         """
-
-        keys = list(d.keys())
-        if choice == "append":
-            for key, val in list(zip(keys, vals)):
-                d[key].append(val)
-        elif choice == "set":
-            for key, val in list(zip(keys, vals)):
-                d[key][idx] = val
+        if not self.paused:
```

Comment:
I don't think this in particular needed to be a looped thread, it may have been simpler to update the status when new data came in e.g. from an epics monitor rather than running through this block at a refresh rate